### PR TITLE
Documentation doxy (#69) - pt. 1

### PR DESCRIPTION
**Pull request by adamdbrw**
Documentation + header cleanup.
- Documentation for Sensor, Lidar, IMU, Frame and QoS
- Header cleanup: ordering and removing unnecesary ones

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

### CODE REVIEW

File: Code/Source/Clock/SimulationClock.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace ROS2
         if (!m_clockPublisher)
         {   //Lazy construct
             auto ros2Node = ROS2Interface::Get()->GetNode();
-
+
```
Comment (lumberyard-employee-dm) [inline Code/Source/Clock/SimulationClock.cpp:43]: nitpick: extra whitespace has been added on blank line

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Source/Frame/ROS2FrameComponent.cpp [hunk 2]
```diff
@@ -52,7 +53,7 @@ namespace ROS2
 
     const char* ROS2FrameComponent::GetGlobalFrameName()
     {
-        // TODO - parametrize this, sometimes it is "world" and sometimes "map".
+        // TODO - parametrize this (typically: "odom", "world" and sometimes "map")
```
Comment (spham-amzn) [inline Code/Source/Frame/ROS2FrameComponent.cpp:56]:
```suggestion
        // TODO - parameterize this (typically: "odom", "world" and sometimes "map")
```

File: Code/Source/Frame/ROS2FrameComponent.h [hunk 3]
```diff
@@ -34,34 +35,41 @@ namespace ROS2
...
 
-        // Transform for direct parent if parent ROS2Frame is found, otherwise transformation is global ("world")
+        //! Get AZ Transform for this frame
+        //! @return If parent ROS2Frame is found, return its Transform.
+        //! Otherwise, return a global Transform.
         const AZ::Transform& GetFrameTransform() const;
 
-        // Returns whatever is set as global frame name in ros2 ecosystem (typically, "map" or "world")
-        static const char* GetGlobalFrameName();
+        //! Global frame name in ros2 ecosystem.
```
Comment (spham-amzn) [inline Code/Source/Frame/ROS2FrameComponent.h:51]:
```suggestion
        //! Get the global frame name in ros2 ecosystem.
```

File: Code/Source/Clock/SimulationClock.cpp (hunk 1 above)
Comment (adamdbrw) [reply Code/Source/Clock/SimulationClock.cpp]: Removed this empty line altogether

**Pull request merged**